br/pkg/lightning/backend/external/byte_reader.go

```diff
@@ -192,24 +191,23 @@ func (r *byteReader) switchToConcurrentReader() error {
 	return nil
 }
 
-// readNBytes reads the next n bytes from the reader and returns a buffer slice containing those bytes.
-// The returned slice (pointer) can not be used after r.reset. In the same interval of r.reset,
-// byteReader guarantees that the returned slice (pointer) will point to the same content
-// though the slice may be changed.
-func (r *byteReader) readNBytes(n int) (*[]byte, error) {
+// readNBytes reads the next n bytes from the reader and returns a buffer slice
+// containing those bytes. The content of returned slice may be changed after
+// next call.
+func (r *byteReader) readNBytes(n int) ([]byte, error) {
 	b := r.next(n)
 	readLen := len(b)
 	if readLen == n {
-		ret := &b
-		r.retPointers = append(r.retPointers, ret)
-		return ret, nil
+		return b, nil
 	}
 	// If the reader has fewer than n bytes remaining in current buffer,
 	// `auxBuf` is used as a container instead.
+	if n > 1024*1024*1024 {
```

Comment:
Why is it `1024*1024*1024`?

Reply:
I think it's a defensive coding, and the author of this line is @wjhuang2016 let's ask him

Reply:
Yes，if there is something wrong we can return an error instead of panic. Just choose a large number.

Reply:
I think it is better to do this outside of `readNBytes()` and log the file name & other info.

Could you change this to `1 * size.GB` instead and add comment for it?